ssl/quic/quic_impl.c

```diff
@@ -1353,65 +1899,66 @@ int ossl_quic_peek(SSL *s, void *buf, size_t len, size_t *bytes_read)
  * -----------
  */
 QUIC_TAKES_LOCK
-static size_t ossl_quic_pending_int(const QUIC_CONNECTION *qc)
+static size_t ossl_quic_pending_int(const SSL *s)
 {
+    QCTX ctx;
     size_t avail = 0;
     int fin = 0;
 
-    if (!expect_quic_conn(qc))
+    if (!expect_quic_with_stream_lock(s, /*remote_init=*/-1, &ctx))
         return 0;
 
-    quic_lock((QUIC_CONNECTION *)qc);
-
-    if (qc->stream0 == NULL || qc->stream0->rstream == NULL)
+    if (ctx.xso->stream == NULL || ctx.xso->stream->rstream == NULL)
         /* Cannot raise errors here because we are const, just fail. */
         goto out;
 
-    if (!ossl_quic_rstream_available(qc->stream0->rstream, &avail, &fin))
+    if (!ossl_quic_rstream_available(ctx.xso->stream->rstream, &avail, &fin))
         avail = 0;
```

Comment:
A bit of paranoia &mdash; e.g. if available fills avail with garbage in some cases before subsequently deciding to fail. I'll take out the initialisation of avail in the declaration instead, since it's needless.